### PR TITLE
Gui

### PR DESCRIPTION
**Pull request by YCGreen**

### CODE REVIEW

File: src/main/java/green/rijks/PaintingComponent.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+//interface adapted from https://www.baeldung.com/java-multiple-keys-map
+//maybe a bad idea let's find out
+interface MultiKeyWrapper {}
+
+record IntegerMultiKeyWrapper(Integer value) implements MultiKeyWrapper {}
+record StringMultiKeyWrapper(String value) implements MultiKeyWrapper {} //use for query requests
+
+public class PaintingComponent extends JComponent {
+    static RijksService service;
+    static ApiKey key;
```
Comment (andrewoid) [inline src/main/java/green/rijks/PaintingComponent.java:27]: I will fail you if you use static variables

File: src/main/java/green/rijks/PaintingComponent.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import java.awt.*;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+
+//interface adapted from https://www.baeldung.com/java-multiple-keys-map
+//maybe a bad idea let's find out
+interface MultiKeyWrapper {}
```
Comment (andrewoid) [inline src/main/java/green/rijks/PaintingComponent.java:20]: You're making this too complicated

File: src/main/java/green/rijks/RijksFrame.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            }
+        }; */
+
+      //  search.getDocument().addDocumentListener(docListener);
+
+
+    }
+}
+
+class Main {
```
Comment (andrewoid) [inline src/main/java/green/rijks/RijksFrame.java:63]: Why did you remove the file Main only to create this class?

Comment (YCGreen) [reply src/main/java/green/rijks/RijksFrame.java:63]: Because it seemed extraneous to have a separate file for one line of code directly related to this file... is there a difference?

Also, if my tests pass for getPage getField etc work, why is my disposable null on calling these methods?

Comment (andrewoid) [reply src/main/java/green/rijks/RijksFrame.java:63]: DIsposable isn't null.
Show me the NullPointerException you are getting
Also make sure that your apikey.properties file is location in `main/java/resources` and not `test/java/resources`

**Comment by YCGreen** [comment]
The NullPointerException is from trying to iterate through the Label[] label on line 34 in RijksFrame, so the PaintingComponent getBlankRequest is returning a null array, coming from sendBlankRequest() which upon closer inspection is sending a perfectly good Artobjects to setUrls, which is for some reason returning a null array.  I went through each iteration of the for loop and it took me about 6 minutes to get to iteration 7ish since it kept stalling at lines 63 and 66 and around 7 it stalled for about 2 minutes before I just stopped the debugger. (Running it normally takes a few seconds before I get the array nullpointerexception.)

Apikey.properties is in main.

**Comment by andrewoid** [comment]
> The NullPointerException is from trying to iterate through the Label[] label on line 34 in RijksFrame, so the PaintingComponent getBlankRequest is returning a null array, coming from sendBlankRequest() which upon closer inspection is sending a perfectly good Artobjects to setUrls, which is for some reason returning a null array. I went through each iteration of the for loop and it took me about 6 minutes to get to iteration 7ish since it kept stalling at lines 63 and 66 and around 7 it stalled for about 2 minutes before I just stopped the debugger. (Running it normally takes a few seconds before I get the array nullpointerexception.) Apikey.properties is in main.

I stand by my other comment, you are making this too complicated.

**Comment by andrewoid** [comment]
Please close this PR (do not merge it)
